src/gui/gui_message_settings_window.cpp

```diff
@@ -7,16 +7,20 @@ void message_log_text::on_update(sys::state& state) noexcept {
 	if(index < int32_t(messages.size())) {
 		auto m = messages[index];
 		auto container = text::create_endless_layout(state, internal_layout,
-				text::layout_parameters{0, 0, base_data.size.x, base_data.size.y, base_data.data.text.font_handle, 0,
-						text::alignment::left, text::text_color::white, false});
+			text::layout_parameters{0, 0, base_data.size.x, base_data.size.y, base_data.data.button.font_handle, 0,
+			text::alignment::left, text::text_color::white, false});
 
 		auto box = text::open_layout_box(container);
-		std::string tags_string = std::string("@") + nations::int_to_tag(state.world.national_identity_get_identifying_int(state.world.nation_get_identity_from_identity_holder(m.source)));
+		std::string source_tag = std::string("@") + nations::int_to_tag(state.world.national_identity_get_identifying_int(state.world.nation_get_identity_from_identity_holder(m.source)));
+		text::add_unparsed_text_to_layout_box(state, container, box, std::string_view{ source_tag });
```

Comment:
you can just add the flag icon directly to the layout box via one of the add to layout box functions. You don't need to make text for it first